From 9d8b070982403ff8e608d4c7cce313bc2a7dce4a Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Thu, 19 Sep 2019 13:54:53 +0200
Subject: [PATCH] util: Blacklist NOTIFY_SOCKET

The NOTIFY_SOCKET environment variable was leaking into systemd
managed GNOME sessions and breaking things like OCI container runtimes
(eg., runc and crun) [1]. This variable is absent in non-systemd
managed sessions.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1753328

https://gitlab.gnome.org/GNOME/gnome-session/merge_requests/22
---
 gnome-session/gsm-util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gnome-session/gsm-util.c b/gnome-session/gsm-util.c
index f6ddefc374dc..1528723cc7d8 100644
--- a/gnome-session/gsm-util.c
+++ b/gnome-session/gsm-util.c
@@ -36,6 +36,7 @@ static gchar *_saved_session_dir = NULL;
 static gchar **child_environment;
 
 static const char * const variable_blacklist[] = {
+    "NOTIFY_SOCKET",
     "XDG_SEAT",
     "XDG_SESSION_ID",
     "XDG_VTNR",
-- 
2.20.1

